mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -116,8 +116,7 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
 
     val distinctSplits = splits.distinct
     if (splits.length != distinctSplits.length) {
-      log.warn(s"Some quantiles were identical. Bucketing to ${distinctSplits.length - 1}" +
-        s" buckets as a result.")
+      log.warn(s"Some quantiles were identical. Will return fewer buckets than requested!")
```

Comment:
considering cases when, there are duplicated non-nan values, along with multiple NaNs in the splits, to get the exact number of buckets, will require extra computation and increase complexity which I dont think it worths for a warning msg